chia/consensus/full_block_to_block_record.py

```diff
@@ -88,6 +88,13 @@ def block_to_block_record(
     )
 
 
+def maybe_uint64(val: Optional[int]) -> Optional[uint64]:
```

Comment:
In case we use this somewhere we shouldn't (somewhere where the type is known) it can propagate the constrained type instead of widening it.  Or, maybe this is just a silly thing to bother with in this case.


```suggestion
@overload
def maybe_uint64(val: None) -> None:
    ...


@overload
def maybe_uint64(val: int) -> uint64:
    ...


def maybe_uint64(val: Optional[int]) -> Optional[uint64]:
```

Reply:
this function is meant to *maybe* construct a `uint64` object, around an existing optional `int`.
It's used when the resulting parameter type or field type is `Optional[uint64]` and we don't really want to change it.